Libs/Widgets/ctkConsole.cpp

```diff
@@ -1252,6 +1252,36 @@ void ctkConsole::setRunFileOptions(const RunFileOptions& newOptions)
   d->RunFileAction->setEnabled(newOptions.testFlag(ctkConsole::RunFileShortcut));
 }
 
+//-----------------------------------------------------------------------------
+const QString& ctkConsole::commandBuffer()
+{
+  Q_D(ctkConsole);
+  return d->commandBuffer();
+}
+
+//-----------------------------------------------------------------------------
+void ctkConsole::setCommandBuffer(const QString& command)
+{
+  Q_D(ctkConsole);
+  d->replaceCommandBuffer(command);
+}
+
+//-----------------------------------------------------------------------------
+const QStringList& ctkConsole::commandHistory()
+{
+  Q_D(ctkConsole);
+  return d->CommandHistory;
+}
+
+//-----------------------------------------------------------------------------
+void ctkConsole::setCommandHistory(const QStringList& commandHistory)
+{
+  Q_D(ctkConsole);
+  d->CommandHistory = commandHistory;
+  d->CommandHistory.append("");
```

Comment:
Why is this empty item needed?

Reply:
Yes, looks odd but that's the design.  The empty string is the new command buffer (the new empty command line).  This is the way it's initialized in the constructor.  I figured it was better to add it here rather than making the user of the method have to add an empty string to the list.

Reply:
Strange design, but we don't need to change it right now. Could you just add your explanation above as a comment to the code? Also, to be consistent, we should remove the extra empty string from the returned command history stringlist in `const QStringList& ctkConsole::commandHistory()`.